rocketpy/environment/fetchers.py

```diff
@@ -0,0 +1,433 @@
+
+    Raises
+    ------
+    RuntimeError
+        If there is a problem reaching the Open-Elevation API servers.
+    """
+    print(f"Fetching elevation from open-elevation.com for lat={lat}, lon={lon}...")
+    request_url = (
+        "https://api.open-elevation.com/api/v1/lookup?locations" f"={lat},{lon}"
+    )
+    try:
+        response = requests.get(request_url)
+    except requests.exceptions.RequestException as e:
+        raise RuntimeError("Unable to reach Open-Elevation API servers.") from e
+    results = response.json()["results"]
+    return results[0]["elevation"]
+
+
+@exponential_backoff(max_attempts=5, base_delay=2, max_delay=60)
+def fetch_atmospheric_data_from_windy(lat, lon, model):
+    """Fetches atmospheric data from Windy.com API for a given latitude and
+    longitude, using a specific model.
+
+    Parameters
+    ----------
+    lat : float
+        The latitude of the location.
+    lon : float
+        The longitude of the location.
+    model : str
+        The atmospheric model to use. Options are: ecmwf, GFS, ICON or ICONEU.
+
+    Returns
+    -------
+    dict
+        A dictionary containing the atmospheric data retrieved from the API.
+
+    Raises
+    ------
+    ValueError
+        If an invalid response is received from the API.
+    """
+    model = model.lower()
+    if model[-1] == "u":  # case iconEu
+        model = "".join([model[:4], model[4].upper(), model[5:]])
+
+    url = (
+        f"https://node.windy.com/forecast/meteogram/{model}/{lat}/{lon}/"
+        "?step=undefined"
+    )
+
+    try:
+        response = requests.get(url).json()
+        if "data" not in response.keys():
+            raise ValueError(
+                f"Could not get a valid response for {model} from Windy. "
+                "Check if the coordinates are set inside the model's domain."
+            )
+    except requests.exceptions.RequestException as e:
+        if model == "iconEu":
+            raise ValueError(
+                "Could not get a valid response for Icon-EU from Windy. "
+                "Check if the coordinates are set inside Europe."
+            ) from e
+
+    return response
+
+
+def fetch_gfs_file_return_dataset(max_attempts=10, base_delay=2):
+    """Fetches the latest GFS (Global Forecast System) dataset from the NOAA's
+    GrADS data server using the OpenDAP protocol.
+
+    Parameters
+    ----------
+    max_attempts : int, optional
+        The maximum number of attempts to fetch the dataset. Default is 10.
+    base_delay : int, optional
+        The base delay in seconds between attempts. Default is 2.
+
+    Returns
+    -------
+    netCDF4.Dataset
+        The GFS dataset.
+
+    Raises
+    ------
+    RuntimeError
+        If unable to load the latest weather data for GFS.
+    """
+    time_attempt = datetime.now(tz=timezone.utc)
+    attempt_count = 0
+    dataset = None
+
+    while attempt_count < max_attempts:
```

Comment:
Isn't this functionality provided by the `exponential_backoff` decorator? This repeats in other fetchers, some of which use the mentioned decorator.

Reply:
Your question id good.

I don`t really like this behavior...
But what the code is doing is to actually use the attempt count to find the latest data available on the server.

We should REALLY do this retrieval in another way... But I would leave this for another PR. 

I will leave a TODO in the code so we dont forget.